src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/PropertyPages/DebugPageViewModel.cs

```diff
@@ -506,74 +555,56 @@ private void EnvironmentVariables_ValidationStatusChanged(object sender, EventAr
         /// <summary>
         /// Called whenever the debug targets change. Note that after a save this function will be
         /// called. It looks for changes and applies them to the UI as needed. Switching profiles
-        /// will also cause this to change as the active profile is stored in profiles snaphost.
+        /// will also cause this to change as the active profile is stored in the profiles snapshot.
         /// </summary>
         internal virtual void InitializeDebugTargetsCore(ILaunchSettings profiles)
         {
-            bool profilesChanged = true;
-            bool IISSettingsChanged = true;
+            var newSettings = profiles.ToWritableLaunchSettings();
 
             // Since this get's reentered if the user saves or the user switches active profiles.
-            if (DebugProfiles != null)
+            if (CurrentLaunchSettings != null && !CurrentLaunchSettings.SetttingsDiffer(newSettings))
             {
-                profilesChanged = profiles.ProfilesAreDifferent(DebugProfiles.Select(p => (ILaunchProfile)p).ToList());
-                if (!profilesChanged && !IISSettingsChanged)
-                {
-                    return;
-                }
+                return;
             }
 
             try
             {
-                // This should never change the dirty state
+                // This should never change the dirty state when loading the dialog
                 PushIgnoreEvents();
 
-                if (profilesChanged)
-                {
-                    // Remember the current selection
-                    string curProfileName = SelectedDebugProfile?.Name;
+                // Remember the current selection
+                string curProfileName = SelectedDebugProfile?.Name;
 
-                    // Load debug profiles
-                    var debugProfiles = new ObservableCollection<LaunchProfile>();
+                // Update the set of settings and generate a property change so the list of profiles gets updated. Note that we always
+                // clear the active profile on the CurrentLaunchSettings so that when we do set one and property changed event is set
+                CurrentLaunchSettings = newSettings;
+                CurrentLaunchSettings.ActiveProfile = null;
+                NotifyProfileCollectionChanged();
 
-                    foreach (var profile in profiles.Profiles)
+                // If we have a selection, we want to leave it as is
+                if (curProfileName == null || newSettings.Profiles.FirstOrDefault(p => { return LaunchProfile.IsSameProfileName(p.Name, curProfileName); }) == null)
```

Comment:
This line can be significantly condensed by doing `FirstOrDefault(p => LaunchProfile.IsSameProfileName(p.Name, curProfileName))`, getting rid of the unnecessary brackets, return, and semicolon.